Resolver/Elastos-DID-Resolver-Specification_cn.md

```diff
@@ -402,3 +381,246 @@ JSON-RPC的默认响应属性，字符串值，指定JSON-RPC协议的版本。
   }
 }
 ```
+
+## DID Query
+
+DID Query是用来对ID侧链上的DID文档进行查询的方法，查询的数据范围是当前非deactivate状态的最后版本DID文档，其中包含过期的文档。
+
+### Query Request
+
+#### method
+
+字符串值，其中包含要调用的方法的名称。对DID Query请求，值为"query"。
+
+#### params
+
+对象值，包含在query方法调用中需要使用的参数。参数定义如下：
+
+- service
+
+  可选，表示 service type 的字符串。关于service type，参见亦来云 DID 方法规范中 service 的定义。支持多个类型的或 "|" 以及与 "&", 比如："type1 | type2", "type1 & type2"。
+
+- credential
+
+  可选，表示 credential type 的字符串。关于credential type，参见亦来云可验证声明规范中凭证类型定义。支持多个类型的或 "|" 以及与 "&", 比如："type1 | type2", "type1 & type2"。
+
+- query
+
+  可选，采用 MongoDB 语法的查询条件。
+
+上述三个参数虽然都可选，但同时只能支持一个，并且必须有一个。
+
+#### id
+
+客户端设定的请求标识符，必须包含字符串，数字或NULL值（如果包含）。 值通常不应为null，并且数字不应包含小数部分。
+
+### Resolve Response
+
+#### jsonrpc
+
+JSON-RPC的默认响应属性，字符串值，指定JSON-RPC协议的版本。 必须为“ 2.0”。
+
+#### result
+
+如果query执行成功，则包含此成员，值为query的结果对象。如果调用query时出错，则该成员不存在或者为null。DID Resolver解析结果对象的定义参考[Query result object](#query-result-object)。
+
+#### error
+
+如果Resolve执行正常，则该成员不存在或者为null。如果Resolve执行错误，那么包含该成员，值为错误信息对象。错误信息对象的定义参考[Error object](#error-object)。
+
+#### id
+
+此成员是必需的。它必须与请求对象中id成员的值相同。如果在请求对象中检查ID时发生错误（例如查询错误/无效请求），则该值必须为null。
+
+### Query Result Object
+
+当DID Query成功执行时，response中必须包含result成员，其值是符合查询条件的DID数组。如果没有符合条件的DID，那么返回一个空数组。
+
+### 示例
+
+#### 查询包含特定服务的DID
+
+##### Request
+
+```json
+{
+    "method": "query",
+    "params":{
+        "service": "CredentialRepositoryService & CarrierService",
```

Comment:
Maybe we should not have and or or operators for those simple queries. What about using a simple query field (basic case), and if we need something more complex we use "query", mongo style? Otherwise it's strange, we can sometimes have "and"/"or" query style like mongo, and sometimes like. custom "&" or "|"